### ISSUE TITLE
relative path to junit reporter output file errs

### ISSUE THREAD

**Issue by aronatkins** [r-lib/testthat#1218]
Given a test runner:

```r
# filename: runner.R
library(testthat)

args = commandArgs(trailing = TRUE)
target_directory <- args[1]

options(testthat.output_file = file.path(target_directory, "test-results.xml"))
testthat::test_dir(target_directory, reporter = "junit", stop_on_failure = TRUE)
```

Given a simple test:

```
# filename: test/R/test_something.R
test_that("one is one", {
  expect_equal(1,1)
})
```

Running the `runner.R` errs:

```
Rscript --vanilla ./runner.R test/R
Error: 'test/R' does not exist in current working directory ('/Users/aron/dev/tt/test/R').
Execution halted
```

This issue appeared with the shift to `testthat` 3.0.0; we were using 2.3.2 previously.

**Comment by aronatkins** [comment]
We are working around this problem by converting the target directory into its canonical form with `normalizePath`:

```r
target_directory <- normalizePath(args[1])
```

**Comment by hadley** [comment]
I bet this is because we're setting working directories differently, and need to canonicalise the path earlier.

**Comment by hadley** [comment]
Fixed in 4e7c944097ecf06d4afadef1bf84b9b381ba5c91
